src/init.cpp

```diff
@@ -1261,7 +1261,7 @@ bool AppInitMain(NodeContext& node, interfaces::BlockAndHeaderTipInfo* tip_info)
         for (const std::string& socket_addr : args.GetArgs(port_option)) {
             std::string host_out;
             uint16_t port_out{0};
-            if (!SplitHostPort(socket_addr, port_out, host_out)) {
+            if (socket_addr.rfind("ipc:", 0) != 0 && !SplitHostPort(socket_addr, port_out, host_out)) {
```

Comment:
`rfind` is a very weird choice here. Suggest using [`compare`](https://cplusplus.com/reference/string/string/compare/)

Reply:
`compare` throws an exception if the input string is lower than 4 character, but since this a conflict PR, please close it.